worker/ffmpeg_worker.py

```diff
@@ -43,53 +47,50 @@ def transcode(self, request: FFmpegRequest, context) -> FFmpegResponse:
+                                         ru_maxrss=process.rusage.ru_maxrss,
+                                         ru_ixrss=process.rusage.ru_ixrss,
+                                         ru_idrss=process.rusage.ru_idrss,
+                                         ru_isrss=process.rusage.ru_isrss,
+                                         ru_minflt=process.rusage.ru_minflt,
+                                         ru_majflt=process.rusage.ru_majflt,
+                                         ru_nswap=process.rusage.ru_nswap,
+                                         ru_inblock=process.rusage.ru_inblock,
+                                         ru_oublock=process.rusage.ru_oublock,
+                                         ru_msgsnd=process.rusage.ru_msgsnd,
+                                         ru_msgrcv=process.rusage.ru_msgrcv,
+                                         ru_nsignals=process.rusage.ru_nsignals,
+                                         ru_nvcsw=process.rusage.ru_nvcsw,
+                                         ru_nivcsw=process.rusage.ru_nivcsw)))
+
+
+class Process:
     """
-    with tempfile.TemporaryDirectory() as mount_point:
-        os.chdir(mount_point)
-        try:
-            _mount_buckets(request.buckets)
-            with subprocess.Popen(['ffmpeg', *request.ffmpeg_arguments],
-                                  stdout=subprocess.PIPE,
-                                  stderr=subprocess.STDOUT,
-                                  universal_newlines=True,
-                                  bufsize=1) as pipe:
-                for line in pipe.stdout:
-                    yield line
-        finally:
-            _unmount_buckets(request.buckets)
-
-
-def _mount_buckets(buckets: List[str]) -> None:
-    """Mounts GCS buckets in the current directory.
-
-    Args:
-        buckets: The names of the buckets to mount.
+    Wrapper class around subprocess.Popen class.
+    This class records the resource usage of the terminated process.
     """
-    for bucket in buckets:
-        os.mkdir(bucket)
-        subprocess.run(['gcsfuse', bucket, bucket], check=True)
 
+    def __init__(self, args):
+        self._pipe = subprocess.Popen(args,
+                                      stdout=subprocess.PIPE,
+                                      stderr=subprocess.STDOUT,
+                                      universal_newlines=True,
+                                      bufsize=1)
+        self.returncode = None
+        self.rusage = None
 
-def _unmount_buckets(buckets: List[str]) -> None:
-    """Unmounts GCS buckets in the current directory.
 
-    Args:
-        buckets: The names of the buckets to unmount.
-    """
-    for bucket in buckets:
-        subprocess.run(['fusermount', '-u', bucket], check=True)
+    def __iter__(self):
+        yield from self._pipe.stdout
+        _, status, self.rusage = os.wait4(self._pipe.pid, 0)
+        self.returncode = (os.WEXITSTATUS(status) if os.WIFEXITED(status)
```

Comment:
just return the status directly and let client handle the semantics of dealing with whether it was exited or signaled .